### PR TITLE
Course Search: Change departments filter to "OR"

### LINKED ISSUE #2422: Change Departments filter in Course Search to "OR"
I realized that it probably makes more sense for the departments filter to be an "OR" filter instead of "AND", so that you can select `MATH` and `CSCI` and get all the courses in either of those departments that match your query, rather than all the courses in both of those of those departments that match the query.

### PR DESCRIPTION
**Pull request by hannesmcman**
Closes #2422 .

I had to modify the logic in `applyOrListFilter` to work with filters like departments, where a course can have multiple departments.

### CODE REVIEW

File: source/views/components/filter/apply-filters.js [hunk 1]
```diff
@@ -53,10 +54,16 @@ export function applyListFilter(filter: ListType, item: any): boolean {
...
-	itemValue: string,
+	itemValue: string | string[],
 ): boolean {
 	// An item passes, if its value is in the filter's selected items array
-	return filterValue.map(f => f.title).includes(itemValue)
+	if (!Array.isArray(itemValue)) {
+		return filterValue.map(f => f.title).includes(itemValue)
+	}
+
+	const valueToCheckAgainst = filterValue.map(f => f.title.toString())
```
Comment (hawkrives) [inline source/views/components/filter/apply-filters.js:64]: I don't think we need toString here, do we?

File: source/views/sis/course-search/search.js [hunk 2]
```diff
@@ -144,6 +144,7 @@ class CourseSearchView extends React.PureComponent<Props, State> {
 		const query = text.toLowerCase()
 		this.setState(() => ({query: query}))
 
+		console.log(this.props.allCourses[0])
```
Comment (hawkrives) [inline source/views/sis/course-search/search.js]: Remove debugging :)

**Review by hawkrives — CHANGES_REQUESTED** [review]

File: source/views/components/filter/apply-filters.js (hunk 1 above)
Comment (hannesmcman) [reply source/views/components/filter/apply-filters.js:64]: Flow gets mad without it, since title could be a number

**Comment by hannesmcman** [comment]
Is that jest test failure a problem? Or is the test wrong?

**Comment by hawkrives** [comment]
DST problems

**Comment by hawkrives** [comment]
Just merge master in, push, and the build should go green.

**Review by hawkrives — APPROVED** [review]

**Pull request merged**